YARN-1904. Ensure exceptions thrown in ClientRMService & ApplicationHistoryClientService are uniform when application-attempt is not found. Contributed by Zhijie Shen.

+import org.apache.hadoop.yarn.api.records.ApplicationAttemptId;
+import org.apache.hadoop.yarn.api.records.ContainerId;
+      ApplicationAttemptId appAttemptId = request.getApplicationAttemptId();
-              .getApplicationAttempt(request.getApplicationAttemptId()));
+              .getApplicationAttempt(appAttemptId));
-        throw new ApplicationAttemptNotFoundException(e.getMessage());
+        String msg = "ApplicationAttempt with id '" + appAttemptId +
+            "' doesn't exist in the history store.";
+        LOG.error(msg, e);
+        throw new ApplicationAttemptNotFoundException(msg);
+      ApplicationId applicationId = request.getApplicationId();
-        ApplicationId applicationId = request.getApplicationId();
-        throw new ApplicationNotFoundException(e.getMessage());
+        String msg = "Application with id '" + applicationId +
+            "' doesn't exist in the history store.";
+        LOG.error(msg, e);
+        throw new ApplicationNotFoundException(msg);
+      ContainerId containerId = request.getContainerId();
-            GetContainerReportResponse.newInstance(history.getContainer(request
-              .getContainerId()));
+            GetContainerReportResponse.newInstance(
+                history.getContainer(containerId));
-        throw new ContainerNotFoundException(e.getMessage());
+        String msg = "Container with id '" + containerId +
+            "' doesn't exist in the history store.";
+        LOG.error(msg, e);
+        throw new ContainerNotFoundException(msg);
